psi4/driver/procrouting/proc.py

```diff
@@ -1198,7 +1198,8 @@ def scf_wavefunction_factory(name, ref_wfn, reference, **kwargs):
 
     # Set the DF basis sets
     if (("DF" in core.get_global_option("SCF_TYPE")) or
-            (core.get_option("SCF", "DF_SCF_GUESS") and (core.get_global_option("SCF_TYPE") == "DIRECT"))):
+            (core.get_option("SCF", "DF_SCF_GUESS") and (core.get_global_option("SCF_TYPE") == "DIRECT")) or
+            (core.get_global_option("SCF_TYPE") == "COMPOSITE" and core.get_option("SCF", "J_TYPE") == "DIRECT_DF")):
```

Comment:
I think from a group meeting that arbitrary composites weren't planned -- only specific ones were to be allowed or encouraged? Since dozens of new combinations aren't foreseen, I'd recommend allowable ones be given their own JK scf_type (e.g., `DIRECT_DF_LINK`). 
* user only needs to set one option, not three, to access
* if you need further options for internal routing, that's fine, like https://github.com/psi4/psi4/blob/master/psi4/src/read_options.cc#L3221-L3224
* you won't need to program defensively for ways a user can mess up: e.g., j_type=direct_df, k_type=link, no scf_type so defaults to pk, so now you've got no fitting bases attached
* further downstream, when figuring out if a correlated method can be built atop the scf-composite reference or if a gradient method can be, it'll be easier if describable in one label.

Glad to discuss further, especially if this sounds all wrong for current plans.

Reply:
Ah, that makes sense. I think we can keep the 'JBase' and 'KBase' classes, but instead of composite JK, we can make wrapper JK classes that call `Direct_DF_J` and `LinK`. I think a better solution is to keep the CompositeJK structure, but not allow the user to set `scf_type` to composite. For example, if user sets scf_type to `Direct_DF_LinK`, a Composite JK will be built with direct_df_j and linK

Reply:
Sounds good. I see the benefit in the compositeJK structure in the JK code, so long as the user has simple access.

Reply:
Got rid of explicit calling of `CompositeJK`, to call Direct DF J and LinK, user sets `scf_type` to `DIRECT_DF_LINK`